src/pkg/api/auth/cluster/jwt.go

```diff
@@ -39,16 +47,25 @@ func ValidateJWT(w http.ResponseWriter, r *http.Request) bool {
 		for _, group := range groups {
 			for _, allowedGroup := range allowedGroups {
 				if group == allowedGroup {
-					// Group is allowed
-					return true
+					// Group is allowed, add group and username to requestcontext
+					// todo: handle multiple groups
+					// set context values on request
+					r = r.WithContext(context.WithValue(r.Context(), groupKey, group))
+					if preferredUsername, ok := token.Claims.(jwt.MapClaims)["preferred_username"].(string); ok {
+						r = r.WithContext(context.WithValue(r.Context(), userKey, preferredUsername))
+					} else {
+						http.Error(w, "Invalid token claims", http.StatusUnauthorized)
+						return r, false
+					}
+					return r, true
```

Comment:
nit: this `return r, true` line executes if line 54 is `ok`, and subsequently line 56 does _not_ execute, so in the interest of simplicity and keeping things close to each other since we're returning anyway, we can reduce a layer of `if {} else {}` like so:

```

```suggestion
					if preferredUsername, ok := token.Claims.(jwt.MapClaims)["preferred_username"].(string); ok {
						r = r.WithContext(context.WithValue(r.Context(), userKey, preferredUsername))
					        return r, true
					} 
					http.Error(w, "Invalid token claims", http.StatusUnauthorized)
					return r, false
```

Reply:
updated, appreciate you taking a look!